### PR TITLE
Fix barlow twins loss

### PR DESCRIPTION
**Pull request by IgorSusmelj**
## Changes

The existing implementation of the loss function was very slow (training could take 5x longer) because we used slow operations. This change uses a more efficient way of computing the loss.

### CODE REVIEW

**Comment by IgorSusmelj** [comment]
I closed this PR and added the changes directly to the PR that adds the imagenet benchmark

**Review by MalteEbner — COMMENTED** [review]
While the code looks correct, I am not too sure about it. Can you either

- write a unittest for it?
- ask GPT4 if the two code samples are the same and post the answer?

Furthermore, I would rename the variables:
- rename `on_diag` to `invariance_loss`
- rename `off_diag` to `redundancy_reduction_loss`

These are the terms in the paper:

image
